qiskit/pulse/builder.py

```diff
@@ -1892,89 +1925,159 @@ def snapshot(label: str, snapshot_type: str = "statevector"):
+        each parameter object as a python dictionary. For example,
+
+        .. jupyter-execute::
+
+            amp1 = circuit.Parameter('amp')
+            amp2 = circuit.Parameter('amp')
+
+            with pulse.build() as subroutine:
+                pulse.play(pulse.Gaussian(160, amp1, 40), pulse.DriveChannel(0))
+                pulse.play(pulse.Gaussian(160, amp2, 40), pulse.DriveChannel(1))
+
+            with pulse.build() as pulse_prog:
+                pulse.call(subroutine, value_dict={amp1: 0.1, amp2: 0.3})
+
+            print(pulse_prog)
 
     Args:
         target: Target circuit or pulse schedule to call.
-        name: Name of subroutine if defined.
-        value_dict: Parameter object and assigned value mapping. This is more precise way to
-            identify a parameter since mapping is managed with unique object id rather than
-            name. Especially there is any name collision in a parameter table.
-        kw_params: Parameter values to bind to the target subroutine
-            with string parameter names. If there are parameter name overlapping,
-            these parameters are updated with the same assigned value.
+        name: Optional. Name of subroutine if defined.
+        value_dict: Optional. Parameters assigned to the ``target`` program.
+            If this dictionary is provided, the ``target`` program is copied and
+            then stored in the main built schedule and its parameters are assigned to the given values.
+            This dictionary is keyed on the :class:`~.Parameter` object,
+            thus parameter name collision can be avoided.
+        kw_params: Alternative way to provide parameters.
+            Since this is keyed on the string parameter name,
+            the parameters having the same name are all updated together.
+            If you want to avoid name collision, use ``value_dict`` with :class:`~.Parameter`
+            object instead.
 
     Raises:
         exceptions.PulseError: If the input ``target`` type is not supported.
     """
     if not isinstance(target, (circuit.QuantumCircuit, Schedule, ScheduleBlock)):
-        raise exceptions.PulseError(
-            f'Target of type "{target.__class__.__name__}" is not supported.'
-        )
+        raise exceptions.PulseError(f"'{target.__class__.__name__}' is not valid target object.")
+    _active_builder().call_subroutine(
+        subroutine=target, name=name, value_dict=value_dict, **kw_params
+    )
 
-    _active_builder().call_subroutine(target, name, value_dict, **kw_params)
+
+def refer(*ref_keys: str, name: Optional[str] = None):
+    """Refer to undefined subroutine by string keys.
+
+    A :class:`~qiskit.pulse.instructions.Reference` instruction is implicitly created
+    and schedule can be separately registered to the references at a later stage.
+
+    .. code-block:: python
+
+        from qiskit import pulse
+
+        with pulse.build() as main_prog:
+            pulse.refer("x_gate", "q0")
```

Comment:
Should this be
```suggestion
            pulse.refer(("x_gate", "q0"))
```
?

Reply:
This is variadic argument, so not needed be a tuple here (it is internally converted into tuple).